PepperDashEssentials/PluginLoading/PluginLoading.cs

```diff
@@ -0,0 +1,444 @@
+using System.Linq;
+using System.Text;
+using Crestron.SimplSharp;
+using Crestron.SimplSharp.CrestronIO;
+using Crestron.SimplSharp.Reflection;
+
+using PepperDash.Core;
+using PepperDash.Essentials.Core;
+
+namespace PepperDash.Essentials
+{
+    /// <summary>
+    /// Deals with loading plugins at runtime
+    /// </summary>
+    public static class PluginLoader
+    {
+        /// <summary>
+        /// The complete list of loaded assemblies. Includes Essentials Framework assemblies and plugins
+        /// </summary>
+        public static List<LoadedAssembly> LoadedAssemblies { get; private set; }
+
+        /// <summary>
+        /// The list of assemblies loaded from the plugins folder
+        /// </summary>
+        static List<LoadedAssembly> LoadedPluginFolderAssemblies;
+
+        /// <summary>
+        /// The directory to look in for .cplz plugin packages
+        /// </summary>
+        static string _pluginDirectory = Global.FilePathPrefix + "plugins";
+
+        /// <summary>
+        /// The directory where plugins will be moved to and loaded from
+        /// </summary>
+        static string _loadedPluginsDirectoryPath = _pluginDirectory + Global.DirectorySeparator + "loadedAssemblies";
+
+        // The temp directory where .cplz archives will be unzipped to
+        static string _tempDirectory = _pluginDirectory + Global.DirectorySeparator + "temp";
+
+
+        static PluginLoader()
+        {
+            LoadedAssemblies = new List<LoadedAssembly>();
+            LoadedPluginFolderAssemblies = new List<LoadedAssembly>();
+        }
+
+        /// <summary>
+        /// Retrieves all the loaded assemblies from the program directory
+        /// </summary>
+        public static void AddProgramAssemblies()
+        {
+            Debug.Console(2, "Getting Assemblies loaded with Essentials");
+            // Get the loaded assembly filenames
+            var appDi = new DirectoryInfo(Global.ApplicationDirectoryPathPrefix);
+            var assemblyFiles = appDi.GetFiles("*.dll");
+
+            Debug.Console(2, "Found {0} Assemblies", assemblyFiles.Length);
+
+            foreach (var fi in assemblyFiles)
+            {
+                string version = string.Empty;
+                Assembly assembly = null;
+
+                switch (fi.Name)
+                {
+                    case ("PepperDashEssentials.dll"):
+                        {
+                            version = Global.AssemblyVersion;
+                            assembly = Assembly.GetExecutingAssembly();
+                            break;
+                        }
+                    case ("PepperDash_Core.dll"):
+                        {
+                            version = PepperDash.Core.Debug.PepperDashCoreVersion;
+                            break;
+                        }
+                }
+
+                LoadedAssemblies.Add(new LoadedAssembly(fi.Name, version, assembly));
+            }
+
+            if (Debug.Level > 1)
```

Comment:
Not sure this If statement is needed...unless you're wanting to prevent the loop from running UNLESS someone wants to see it?